pkg/controller/eks/nodegroup/controller.go

```diff
@@ -219,3 +229,70 @@ func (t *tagger) Initialize(ctx context.Context, mg resource.Managed) error {
 	}
 	return errors.Wrap(t.kube.Update(ctx, cr), errKubeUpdateFailed)
 }
+
+func versionHasChanged(cr *manualv1alpha1.NodeGroup, ng *awseks.DescribeNodegroupOutput) bool {
+	if ng.Nodegroup.LaunchTemplate != nil && cr.Spec.ForProvider.LaunchTemplate != nil {
+		if !reflect.DeepEqual(ng.Nodegroup.LaunchTemplate.Version, cr.Spec.ForProvider.LaunchTemplate.Version) {
+			return true
+		}
+	}
+	return !reflect.DeepEqual(ng.Nodegroup.Version, cr.Spec.ForProvider.Version)
+}
+
+func generateInput(ctx context.Context, e *external, cr *manualv1alpha1.NodeGroup, ng *awseks.DescribeNodegroupOutput) (*awseks.UpdateNodegroupVersionInput, error) {
```

Comment:
I think this method should be named `GenerateUpdateNodeGroupVersionInput` and should live in `pkg/clients/eks/nodegroup.go`. Like this it sticks to the same pattern as `eks.GenerateUpdateNodeGroupConfigInput`.